### PR TITLE
userdiff: support enum keyword in PHP hunk header

### PR DESCRIPTION
**Pull request by zonuexe**
"enum" keyword will be introduced in PHP 8.1.
https://wiki.php.net/rfc/enumerations

### CODE REVIEW

File: t/t4018/php-enum [hunk 1]
```diff
@@ -0,0 +1,3 @@
+enum RIGHT: string {
+    case Foo = 'ChangeMe';
```
Comment (dscho) [inline t/t4018/php-enum:2]: This seems to break the test suite: https://github.com/git/git/pull/1082/checks?check_run_id=3465586494#step:5:3769

Comment (zonuexe) [reply t/t4018/php-enum:3]: Fixed at 1fe4aa4e692c099e5d7098b1e4076482b7e18eb4 (squashed)

**Review by mohsin1980 — COMMENTED** [review]
https://ghcr.io/v2/homebrew/core/git-remote-hg/blobs/sha256:3903ddefc5ed6142943aa33ba298ac51d054159f0c401bcde044934494202a19

**Pull request closed without merging**